c_glib/arrow-glib/table-builder.cpp

```diff
@@ -155,11 +155,10 @@ garrow_record_batch_builder_new(GArrowSchema *schema, GError **error)
 {
   auto arrow_schema = garrow_schema_get_raw(schema);
   auto memory_pool = arrow::default_memory_pool();
-  std::unique_ptr<arrow::RecordBatchBuilder> arrow_builder;
-  auto status = arrow::RecordBatchBuilder::Make(arrow_schema,
-                                                memory_pool,
-                                                &arrow_builder);
-  if (garrow_error_check(error, status, "[record-batch-builder][new]")) {
+  auto builder_result = arrow::RecordBatchBuilder::Make(arrow_schema, memory_pool);
+
+  if (garrow::check(error, builder_result, "[record-batch-builder][new]")) {
+    std::unique_ptr<arrow::RecordBatchBuilder> arrow_builder = std::move(builder_result).ValueOrDie();
```

Comment:
nit: formatting, as is this change related?

Reply:
It's not related. I've moved this change to #13356 and will rebase after that's merged.